applications/hotornot/src/hon_trie.erl

```diff
@@ -78,14 +86,21 @@ handle_call({'match_did', DID}, _From, State) ->
         'error' -> {'reply', {'error', 'not_found'}, State};
         {'ok', Prefix, RateIds} -> {'reply', {'ok', {Prefix, RateIds}}, State}
     end;
+handle_call('rebuild', _From, State) ->
+    PidRef = spawn_monitor(?MODULE, 'build_trie', [self()]),
```

Comment:
If you spawn_monitor/3, the gen_listener will receive the 'DOWN' message in handle_info/2. I would keep the PidRef around in the server. Maybe `{CurrentTtrie, RebuildPidRef}`

Then handling the rebuilt trie or the DOWN can make sure its the spawned process that sent the message. This could also prevent or provide ways to handle if two or more 'rebuild' commands have been triggered. Like, if State is `{Trie, {Pid, Ref}}` already, ignore a new rebuild command. Or something along those lines.